src/librbd/Utils.cc

```diff
@@ -242,5 +248,142 @@ int get_config_key(librados::Rados& rados, const std::string& uri,
   return 0;
 }
 
+const std::vector<std::string> UNIQUE_PEER_CONFIG_KEYS {
+  {"monmap", "mon_host", "mon_dns_srv_name", "key", "keyfile", "keyring"}};
+
+int init_rados(CephContext *cct,
+               const std::string &cluster_name,
+               const std::string &client_name,
+               const std::string &mon_host,
+               const std::string &key,
+               const std::string &description,
+               const std::vector<const char*>& args,
+               librados::Rados **rados_ref,
+               bool strip_cluster_overrides,
+               const std::string &admin_socket,
+               int preinit_flags) {
+  // NOTE: manually bootstrap a CephContext here instead of via
+  // the librados API to avoid mixing global singletons between
+  // the librados shared library and the daemon
+  // TODO: eliminate intermingling of global singletons within Ceph APIs
+  CephInitParameters iparams(CEPH_ENTITY_TYPE_CLIENT);
+  if (client_name.empty() || !iparams.name.from_str(client_name)) {
```

Comment:
Allow `client_name` to default to `client.admin` implicitly:
```suggestion
  if (!client_name.empty() && !iparams.name.from_str(client_name)) {
```